torch/csrc/distributed/c10d/TraceUtils.h

```diff
@@ -320,11 +320,12 @@ inline std::string get_python_cpp_trace() {
   const auto& s_tb = s_tbs.tracebacks.at(0);
   constexpr auto TB_FMT_CSTR = FMT_COMPILE("#{} {} from {}:{}\n");
   fmt::memory_buffer buf;
+  auto buf_iter = std::back_inserter(buf);
   for (auto idx : c10::irange(s_tb.size())) {
     auto frame_id = s_tb[idx];
     const auto& frame = s_tbs.all_frames.at(frame_id);
     fmt::format_to(
-        std::back_inserter(buf),
+        buf_iter,
```

Comment:
Don't do that because buf_iter keeps changing in the for loop

Reply:
> Don't do that because `std::back_inserter(buf)` keeps changing in the for loop. After the first iteration, it appends to the wrong place.

Oh really, can you provide a source? In the other recent libfmt PR, we did this and it seemed to work fine and pass tests, but I want to make sure this is actually the case. I haven't been able to find documentation that it's bad unless the underlying buffer is invalidated?

Aaron

Reply:
Sorry, I confused it with std::rbegin. Looks like your usage have no problem.